include/clap/factory/draft/preset-discovery.h

```diff
@@ -48,12 +48,24 @@
 // Use it to retrieve const clap_preset_discovery_factory_t* from
 // clap_plugin_entry.get_factory()
 static const CLAP_CONSTEXPR char CLAP_PRESET_DISCOVERY_FACTORY_ID[] =
-   "clap.preset-discovery-factory/draft-1";
+   "clap.preset-discovery-factory/draft-2";
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+enum clap_preset_discovery_location_kind {
+   // The preset are located in a file on the OS filesystem.
+   // The location is then a path which works with the OS file system functions (open, stat, ...)
+   // So both '/' and '\' shall work on Windows as a separator.
+   CLAP_PRESET_DISCOVERY_LOCATION_FILE = 0,
+
+   // The preset is bundled within the plugin DSO itself.
+   // The location must then be null, as the preset are within the plugin itsel and then the plugin
+   // will act as a preset container.
+   CLAP_PRESET_DISCOVERY_LOCATION_PLUGIN = 1,
+};
```

Comment:
When reading the documentation, I think that it might be better to have two location struct types:

- `clap_preset_discovery_file_location_t`
- `clap_preset_discovery_plugin_location_t`

The `kind` member wouldn't be needed anymore.
The `clap_preset_discovery_plugin_location_t` would be without the `location` member.
For declaration there could be two separate methods:

- `declare_file_location`
- `declare_plugin_location`

I think it would prevent some of the clutter of the current documentation.

Reply:
Yes you can write a .clap file which only contains a preset provider, and targets other plugins.

Reply:
We can have additional location kind in the future maybe.
This needs to be flexible.
We could still go for a struct approach, but it'd be something like the events where you need a cast.

Reply:
> Yes you can write a .clap file which only contains a preset provider, and targets other plugins.

Ok, understand.

> We can have additional location kind in the future maybe.

Yes, maybe online preset resources.

> This needs to be flexible. We could still go for a struct approach, but it'd be something like the events where you need a cast.

Understand as well. Conversation resolved, I'd say ;)